VelocidiSDK/VelocidiSDK/VSDKRequest.m

```diff
@@ -0,0 +1,63 @@
+@import Foundation;
+#import "VSDKRequest.h"
+#import "VSDKUtil.h"
+
+@implementation VSDKRequest
+
+- (instancetype)initWithHTTPSessionManager:(AFHTTPSessionManager *)manager{
+    if (self = [self init]) {
+        _manager = manager;
+    }
+    return self;
+}
+
+- (instancetype)init {
+    if (self = [super init]) {
+        _manager = [AFHTTPSessionManager manager];
+        _util = [[VSDKUtil alloc] init] ;
+        [VSDKUtil setAcceptAllResponses:_manager];
+    }
+    return self;
+}
+
+- (void)performRequest: (void (^)(NSURLResponse *response, id responseObject))onSuccessBlock
+        :(void (^)(NSError *error))onFailureBlock {
+    if (![self.util isAdvertisingTrackingEnabled]) {
+        return;
```

Comment:
I think just returning here doesn't give the caller a proper signal that tracking is disabled, particularly because the only interaction callers have with this method is through callbacks. What do you think of calling the `onFailureBlock` with a custom error?

Reply:
Yes, I agree. And this also improves the way we are testing if the request is made. Improved in  e7fd2f6